src/main/java/com/google/devtools/build/lib/bazel/bzlmod/IndexRegistry.java

```diff
@@ -140,10 +150,17 @@ private Optional<byte[]> doGrabFile(
     } else {
       checksum = Optional.empty();
     }
+    if (knownFileHashesMode == KnownFileHashesMode.ENFORCE) {
```

Comment:
`&& useChecksum`? Or wouldn't this always throw for metadata.json when `--lockfile_mode=error`?